### PR TITLE
additions for top related analyses

### PR DESCRIPTION
**Pull request by nfaltermann**

### CODE REVIEW

File: code_generation/subset_template.cxx [hunk 1]
```diff
@@ -14,15 +14,15 @@
...
 #include "include/scalefactors.hxx"
+#include "include/topreco.hxx"
 #include "include/triggers.hxx"
 #include "include/utility/Logger.hxx"
 #include <ROOT/RLogger.hxx>
 #include <TFile.h>
 #include <TTree.h>
 #include <regex>
 #include <string>
+ROOT::RDF::RNode{subsetname}(ROOT::RDF::RNode df0) {
```
Comment (harrypuuter) [inline code_generation/subset_template.cxx:25]: spaces are needed here: 

```
ROOT::RDF::RNode {subsetname} (ROOT::RDF::RNode df0) {
```

**Review by harrypuuter — CHANGES_REQUESTED** [review]
a few smaller remarks. see above :+1:

File: src/jets.cxx [hunk 2]
```diff
@@ -572,6 +635,33 @@ ROOT::RDF::RNode btagValue(ROOT::RDF::RNode df, const std::string &outputname,
...
+
+ROOT::RDF::RNode flavor(ROOT::RDF::RNode df, const std::string &outputname,
+                        const std::string &flavorcolumn,
+                        const std::string &jetcollection, const int &position) {
+    return df.Define(outputname,
+                     [position](const ROOT::RVec<int> &flavorvalue,
+                                const ROOT::RVec<int> &jetcollection) {
+                         int flavorValue = default_int;
+                         try {
+                             const int index = jetcollection.at(position);
```
[the remark this answers is not in the document]
Comment (nfaltermann) [reply src/jets.cxx]: thanks for the suggestion, changed accordingly

File: src/topreco.cxx [hunk 3]
```diff
@@ -0,0 +1,1987 @@
+#ifndef GUARD_TOPRECO_H
+#define GUARD_TOPRECO_H
+
```
[the remark this answers is not in the document]
Comment (nfaltermann) [reply src/topreco.cxx:3]: added

File: code_generation/subset_template.cxx (hunk 1 above)
Comment (nfaltermann) [reply code_generation/subset_template.cxx]: clang -.-

**Review by harrypuuter — APPROVED** [review]
thats a 👍 from me

**Comment by harrypuuter** [comment]
there are still two python files where black is complaining, the error in the build test is unrelated, so I guess we just ignore it :D

**Pull request merged**